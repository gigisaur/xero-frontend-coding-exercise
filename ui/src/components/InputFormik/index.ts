import { FormikProps } from "formik";

type InputType = "text" | "number";

export interface IInputFormikProps {
    formikProps: FormikProps<any>;
    label: string;
    name: string;
    required?: boolean;
    type: InputType;
}

export * from "./InputFormik";
