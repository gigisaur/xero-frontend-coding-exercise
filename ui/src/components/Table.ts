import styled from "styled-components";
import { Color } from "../theme";

export const Table = styled.table`
    border-collapse: collapse;
    width: 100%;
    border: 1px solid ${Color.TableBorder};
`;

export const Td = styled.td`
    padding: 0.5rem;
    color: ${Color.DefaultText};
    font-size: 0.9rem;
    border-bottom: 1px solid ${Color.TableBorder};
`;

export const Th = styled.th`
    padding: 0.5rem;
    border-bottom: 1px solid ${Color.Border};
    color: ${Color.DefaultText};
    opacity: 0.75;
    font-size: 0.8rem;
    text-transform: uppercase;
    text-align: left;
`;

export const Tr = styled.tr`
    &:nth-child(odd) {
        background-color: ${Color.TableBorder};
    }
`;
