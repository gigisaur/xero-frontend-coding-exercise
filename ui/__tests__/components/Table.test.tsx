import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import "jest-styled-components";
import * as React from "react";
import { Table, Td, Th, Tr } from "../../src/components/Table";

test(`When rendered then the HTML matches this snapshot`, () => {
    const { container, unmount } = render(
        <Table>
            <thead>
                <tr>
                    <Th>Header 1</Th>
                    <Th>Header 2</Th>
                </tr>
            </thead>
            <tbody>
                <Tr>
                    <Td>Row 1 Cell 1</Td>
                    <Td>Row 1 Cell 2</Td>
                </Tr>
                <Tr>
                    <Td>Row 2 Cell 1</Td>
                    <Td>Row 2 Cell 2</Td>
                </Tr>
            </tbody>
        </Table>);

    expect(container).toMatchSnapshot();
    unmount();
});

test(`When rendered then there are no accessibility violations`, async () => {
    const { container, unmount } = render(
        <Table>
            <thead>
                <tr>
                    <Th>Header 1</Th>
                    <Th>Header 2</Th>
                </tr>
            </thead>
            <tbody>
                <Tr>
                    <Td>Row 1 Cell 1</Td>
                    <Td>Row 1 Cell 2</Td>
                </Tr>
                <Tr>
                    <Td>Row 2 Cell 1</Td>
                    <Td>Row 2 Cell 2</Td>
                </Tr>
            </tbody>
        </Table>
    );
    const results = await axe(container)

    expect(results).toHaveNoViolations();
    unmount();
});
