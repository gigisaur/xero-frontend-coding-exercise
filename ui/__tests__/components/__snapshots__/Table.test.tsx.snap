// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When rendered then the HTML matches this snapshot 1`] = `
.c0 {
  border-collapse: collapse;
  width: 100%;
  border: 1px solid #EBF6FF;
}

.c3 {
  padding: 0.5rem;
  color: #191919;
  font-size: 0.9rem;
  border-bottom: 1px solid #EBF6FF;
}

.c1 {
  padding: 0.5rem;
  border-bottom: 1px solid #DAE5EC;
  color: #191919;
  opacity: 0.75;
  font-size: 0.8rem;
  text-transform: uppercase;
  text-align: left;
}

.c2:nth-child(odd) {
  background-color: #EBF6FF;
}

<div>
  <table
    class="c0"
  >
    <thead>
      <tr>
        <th
          class="c1"
        >
          Header 1
        </th>
        <th
          class="c1"
        >
          Header 2
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          Row 1 Cell 1
        </td>
        <td
          class="c3"
        >
          Row 1 Cell 2
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          Row 2 Cell 1
        </td>
        <td
          class="c3"
        >
          Row 2 Cell 2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
